include/ddc/kernels/splines/bsplines_non_uniform.hpp

```diff
@@ -108,93 +119,229 @@ class NonUniformBSplines : detail::NonUniformBSplinesBase
+         *
+         * Returns the coordinate of the knot corresponding to the given index. The domain
+         * over which the B-splines are defined is comprised of ncells+1 knots however there are a total of
+         * ncells+1+2*degree knots. The additional knots which control the shape of the B-splines near the
+         * boundary are added before and after the break points. The knot index is therefore in the interval [-degree, ncells+degree]
+         *
+         * @param[in] knot_idx Integer identifying index of the knot.
+         * @return Coordinate of the knot.
+         */
         KOKKOS_INLINE_FUNCTION ddc::Coordinate<Tag> get_knot(int knot_idx) const noexcept
         {
             return ddc::coordinate(ddc::DiscreteElement<mesh_type>(knot_idx + degree()));
         }
 
+        /** @brief Returns the coordinate of the first support knot associated to a DiscreteElement identifying a B-spline.
+         * 
+         * Each B-spline has a support defined over (degree+2) knots. For a B-spline identified by the
+         * provided DiscreteElement, this function returns the first knot in the support of the B-spline.
+         * In other words it returns the lower bound of the support.
+         *
+         * @param[in] ix DiscreteElement identifying the B-spline.
+         * @return Coordinate of the knot.
+         */
         KOKKOS_INLINE_FUNCTION ddc::Coordinate<Tag> get_first_support_knot(
                 discrete_element_type const& ix) const
         {
             return ddc::coordinate(ddc::DiscreteElement<mesh_type>(ix.uid()));
         }
 
+        /** @brief Returns the coordinate of the last support knot associated to a DiscreteElement identifying a B-spline.
+         *
+         * Each B-spline has a support defined over (degree+2) knots. For a B-spline identified by the
+         * provided DiscreteElement, this function returns the last knot in the support of the B-spline.
+         * In other words it returns the upper bound of the support.
+         *
+         * @param[in] ix DiscreteElement identifying the B-spline.
+         * @return Coordinate of the knot.
+         */
         KOKKOS_INLINE_FUNCTION ddc::Coordinate<Tag> get_last_support_knot(
                 discrete_element_type const& ix) const
         {
             return ddc::coordinate(ddc::DiscreteElement<mesh_type>(ix.uid() + degree() + 1));
         }
 
+        /** @brief Returns the coordinate of the first knot in the support of the identified B-spline.
+         *
+         * Each B-spline has a support defined over (degree+2) knots. For a B-spline identified by the
+         * provided DiscreteElement, this function returns the (n+1)-th knot in the support of the B-spline.
+         *
+         * @param[in] ix DiscreteElement identifying the B-spline.
+         * @param[in] n Integer indexing a knot in the support of the B-spline.
+         * @return Coordinate of the knot.
+         */
```

Comment:
Sorry I seem to have given you the wrong brief description here.

```suggestion
        /** @brief Returns the coordinate of the (n+1)th knot in the support of the identified B-spline.
         *
         * Each B-spline has a support defined over (degree+2) knots. For a B-spline identified by the
         * provided DiscreteElement, this function returns the (n+1)-th knot in the support of the B-spline.
         *
         * @param[in] ix DiscreteElement identifying the B-spline.
         * @param[in] n Integer indexing a knot in the support of the B-spline.
         * @return Coordinate of the knot.
         */
```